public/app/plugins/panel/alertlist/module.tsx

```diff
@@ -255,14 +255,19 @@ const unifiedAlertList = new PanelPlugin<UnifiedAlertListOptions>(UnifiedAlertLi
       defaultValue: null,
       editor: function RenderDatasourcePicker(props) {
         return (
-          <DataSourcePicker
-            {...props}
-            type={['prometheus', 'loki', 'grafana']}
-            noDefault
-            current={props.value}
-            onChange={(ds: DataSourceInstanceSettings) => props.onChange(ds.name)}
-            onClear={() => props.onChange(null)}
```

Comment:
I'm a bit confused, is this perhaps a regression in the new data source picker? Passing `onClear` should make the `Select` component "clearable".

https://github.com/grafana/grafana/blob/ced9fcb94e979f519e113b79da636c6cb71c9683/packages/grafana-runtime/src/components/DataSourcePicker.tsx#L181

Reply:
You're right @gillesdemey , and this regression was introduced [here](https://github.com/grafana/grafana/pull/70609/files#diff-2e980068add69b99836bfe6b81a2d79ce742304bdfea8dbcf689d252adda93d9L3).
 I tested locally with the DataSourcePicker we were using before and the clear was working correctly. 
 
<img width="404" alt="Screenshot 2023-11-10 at 09 12 59" src="https://github.com/grafana/grafana/assets/33540275/4dddc614-d3d2-4b75-92cd-704888026ab4">

Reply:
Right, the new datasource picker doesn't care about the `onClear` function https://github.com/grafana/grafana/blob/main/public/app/features/datasources/components/picker/DataSourceDropdown.tsx#L197